{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import os\n",
    "import scvi\n",
    "import scipy\n",
    "import pickle\n",
    "import anndata\n",
    "import logging\n",
    "import warnings\n",
    "import matplotlib\n",
    "import celltypist\n",
    "import scipy.stats\n",
    "import scanpy as sc\n",
    "import scrublet as scr\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "\n",
    "from matplotlib import pyplot as plt\n",
    "from matplotlib import rcParams\n",
    "from matplotlib.legend import Legend\n",
    "import matplotlib.gridspec as gridspec\n",
    "\n",
    "sc.set_figure_params(scanpy=True, dpi=300, dpi_save=300, frameon=True, vector_friendly=True, fontsize=8, \n",
    "                         color_map='Dark2', format='pdf', transparent=True, ipython_format='png2x')\n",
    "\n",
    "warnings.simplefilter(action=\"ignore\", category=FutureWarning)\n",
    "\n",
    "scvi.settings.seed = 94705\n",
    "\n",
    "%config InlineBackend.print_figure_kwargs={'facecolor' : \"w\"}\n",
    "%config InlineBackend.figure_format='retina'\n",
    "\n",
    "rcParams.update({'font.size': 8})\n",
    "rcParams.update({'font.family': 'Helvetica'})\n",
    "rcParams['pdf.fonttype'] = 42\n",
    "rcParams['ps.fonttype'] = 42\n",
    "rcParams['svg.fonttype'] = 'none'\n",
    "rcParams['figure.facecolor'] = (1,1,1,1)\n",
    "\n",
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\"\n",
    "\n",
    "import populationfunctions as pf\n",
    "import dgexfunctions as dgex\n",
    "import generalfunctions as gf\n",
    "\n",
    "logger = logging.getLogger()\n",
    "\n",
    "def gex_clustermap(tmp=None, GOI=None, groupby=None, use_raw=False,\n",
    "                   agg_function='mean',\n",
    "                   genes_on='rows',\n",
    "                   cluster_method='complete',\n",
    "                   row_cluster=True, col_cluster=True,\n",
    "                   row_order=None, col_order=None,\n",
    "                   standard_scale=False, zscore=True,\n",
    "                   dendrogram_ratio=None,\n",
    "                   cmap=None,\n",
    "                   cbar=True, cbar_shrink=1.0, cbar_title='',\n",
    "                   xlabel=None, ylabel=None, \n",
    "                   x_rotation=0, y_rotation=0,\n",
    "                   xtick_fontsize=2, ytick_fontsize=2,\n",
    "                   vmin=-3, vmax=3, figsize=None, fontsize=2):\n",
    "    \n",
    "    \"\"\"\n",
    "    Clustered heatmap for gene expression, with genes requested in GOI [list] on the\n",
    "    genes_on axis and groupby categories on the opposite axis. Expression values represent\n",
    "    the expression in each category using agg_function (mean, median, min, or max).\n",
    "\n",
    "    Returns fig\n",
    "    \"\"\"\n",
    "\n",
    "    if GOI is None:\n",
    "        GOI = tmp.var_names\n",
    "    if figsize is None:\n",
    "        figsize = (2,3)\n",
    "    if dendrogram_ratio is None:\n",
    "        dendrogram_ratio = (0.3,0.05)\n",
    "    if xlabel is None:\n",
    "        xlabel = ''\n",
    "    if ylabel is None:\n",
    "        ylabel = ''\n",
    "    \n",
    "    if use_raw:\n",
    "        df = pd.DataFrame(tmp[:,GOI].raw.X.toarray(),index=tmp.obs.index,columns=GOI)\n",
    "    else:\n",
    "        df = pd.DataFrame(tmp[:,GOI].X.toarray(),index=tmp.obs.index,columns=GOI)\n",
    "        \n",
    "    df = df.merge(tmp.obs, how='left', left_index=True, right_index=True)\n",
    "\n",
    "    if agg_function == 'mean':\n",
    "        df = df.groupby(groupby).mean(numeric_only=True)[GOI]\n",
    "    elif agg_function == 'median':\n",
    "        df = df.groupby(groupby).median(numeric_only=True)[GOI]\n",
    "    elif agg_function == 'min':\n",
    "        df = df.groupby(groupby).min(numeric_only=True)[GOI]\n",
    "    elif agg_function == 'max':\n",
    "        df = df.groupby(groupby).max(numeric_only=True)[GOI]\n",
    "    else:\n",
    "        print(f\"Unknown agg_function {agg_function}.\")\n",
    "        return None\n",
    "\n",
    "    norm_dim = 1\n",
    "    if genes_on=='rows':\n",
    "        df = df.T\n",
    "        norm_dim = 0\n",
    "        \n",
    "    if row_order is not None:\n",
    "        df = df.loc[row_order,:].copy()\n",
    "    if col_order is not None:\n",
    "        df = df.loc[:,col_order].copy()\n",
    "    \n",
    "    if not cbar:\n",
    "        cbar_pos = None\n",
    "    else:\n",
    "        cbar_pos = (-0.02, 0.92, 0.05*cbar_shrink, 0.18*cbar_shrink)\n",
    "    \n",
    "    cbar_kws = dict(ticks=[vmin, 0.50, vmax], orientation='horizontal')\n",
    "\n",
    "    if cmap is None:\n",
    "        if zscore:\n",
    "            cmap = 'RdBu_r'\n",
    "        else:\n",
    "            cmap = 'Reds'\n",
    "\n",
    "    if standard_scale:\n",
    "        g = sns.clustermap(df,\n",
    "                           method= cluster_method,\n",
    "                           standard_scale= norm_dim,\n",
    "                           row_cluster= row_cluster,\n",
    "                           col_cluster= col_cluster,\n",
    "                           dendrogram_ratio= dendrogram_ratio,\n",
    "                           cmap= cmap,\n",
    "                           vmin= vmin,\n",
    "                           vmax= vmax,\n",
    "                           cbar_kws= cbar_kws,\n",
    "                           cbar_pos= cbar_pos,\n",
    "                           xticklabels= df.columns,\n",
    "                           yticklabels= df.index,\n",
    "                           figsize= figsize)\n",
    "    elif zscore:\n",
    "        g = sns.clustermap(df,\n",
    "                           method= cluster_method,\n",
    "                           z_score= norm_dim,\n",
    "                           row_cluster= row_cluster,\n",
    "                           col_cluster= col_cluster,\n",
    "                           dendrogram_ratio= dendrogram_ratio,\n",
    "                           cmap= cmap,\n",
    "                           vmin= vmin,\n",
    "                           vmax= vmax,\n",
    "                           cbar_kws= cbar_kws,\n",
    "                           cbar_pos= cbar_pos,\n",
    "                           xticklabels= df.columns,\n",
    "                           yticklabels= df.index,\n",
    "                           figsize= figsize)\n",
    "    \n",
    "    g.ax_heatmap.set_yticks(np.arange(df.shape[0])+0.5)\n",
    "    g.ax_heatmap.set_xticks(np.arange(df.shape[1])+0.5)\n",
    "    g.ax_heatmap.set_yticklabels(g.ax_heatmap.get_yticklabels(), fontsize=ytick_fontsize, rotation=y_rotation)\n",
    "    g.ax_heatmap.set_xticklabels(g.ax_heatmap.get_xticklabels(), fontsize=xtick_fontsize, rotation=x_rotation)\n",
    "    g.ax_heatmap.set_xlabel(xlabel, fontsize=fontsize)\n",
    "    g.ax_heatmap.set_ylabel(ylabel, fontsize=fontsize)\n",
    "    g.ax_heatmap.tick_params(axis='both', length=1, width=0.5)\n",
    "    g.ax_heatmap.grid(visible=False)\n",
    "    \n",
    "    if cbar:\n",
    "        x0, y0, _w, _h = g.cbar_pos\n",
    "        g.ax_cbar.set_position([x0, y0, g.ax_row_dendrogram.get_position().width*cbar_shrink, 0.02*cbar_shrink])\n",
    "        g.ax_cbar.set_title(cbar_title, fontdict={'fontsize': fontsize/2})\n",
    "        g.ax_cbar.tick_params(axis='x', labelsize=fontsize/4, length=0.5, width=0.5)\n",
    "    \n",
    "    return g"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "h5files = !ls /common/ingn/SCRNA_RAW_DATA/mouse_RTPD/*.h5\n",
    "\n",
    "sample_treatment = {\n",
    "    'B1': 'IgG',\n",
    "    'B2': 'PD1',\n",
    "    'B3': 'PD1',\n",
    "    'B4': 'RTIgG',\n",
    "    'B5': 'RTPD1',\n",
    "    'B6': 'RTPD1',\n",
    "    'C4': 'RTIgG',\n",
    "    'C5': 'RTPD1',\n",
    "    'mA1': 'IgG',\n",
    "    'mB1': 'PD1',\n",
    "    'mC1': 'PD1',\n",
    "    'mD1': 'RTIgG',\n",
    "    'mE1': 'RTPD1',\n",
    "    'mF1': 'RTPD1',\n",
    "}\n",
    "\n",
    "MAX_MITO = 0.1\n",
    "MIN_GENES = 100\n",
    "for f in h5files:\n",
    "    base = f.split('/')[-1]\n",
    "    outf = f.replace(base, base.replace('_filtered_feature_bc_matrix.h5', '.h5ad'))\n",
    "    sample_id = base.split('_')[1]\n",
    "    cohort = 'Cohort1' if 'm' not in sample_id else 'Cohort2'\n",
    "    print(base, outf, sample_id, sample_treatment[sample_id], cohort)\n",
    "    \n",
    "    ad = sc.read_10x_h5(f)\n",
    "    ad.var_names_make_unique()\n",
    "    \n",
    "    ad.obs['sampleID'] = sample_id\n",
    "    ad.obs['cohort'] = cohort\n",
    "    ad.obs['treatment'] = sample_treatment[sample_id]\n",
    "    ad.obs_names = [f'{sample_id}_'+x.split('-')[0] for x in ad.obs_names]\n",
    "    \n",
    "    mito_genes = ad.var_names.str.startswith('mt-')\n",
    "    ad.var_names[mito_genes]\n",
    "    ad.obs['percent_mito'] = np.sum(ad[:, mito_genes].X, axis=1).A1 / np.sum(ad.X, axis=1).A1\n",
    "\n",
    "    sc.pp.calculate_qc_metrics(ad, inplace=True)\n",
    "\n",
    "    max_counts = int(np.quantile(ad.obs.total_counts, 0.99))\n",
    "\n",
    "    passing_barcodes = (ad.obs.query('percent_mito < @MAX_MITO')\n",
    "                        .query('n_genes_by_counts > @MIN_GENES')\n",
    "                        .query('total_counts < @max_counts')\n",
    "                       ).index.tolist()\n",
    "    print(ad.shape, len(passing_barcodes))\n",
    "    \n",
    "    ad = ad[passing_barcodes].copy()\n",
    "    ad.write(outf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import tqdm.auto as tqdm\n",
    "\n",
    "h5adfiles = !ls /common/ingn/SCRNA_RAW_DATA/mouse_RTPD/*.h5ad\n",
    "h5adfiles\n",
    "\n",
    "for h5adf in tqdm.tqdm(h5adfiles):\n",
    "    outf = h5adf.replace('.h5ad', '_scrublet.csv')\n",
    "    print(h5adf, outf)\n",
    "    \n",
    "    ad = sc.read_h5ad(h5adf)\n",
    "    sc.pp.normalize_total(ad, target_sum=10000)\n",
    "    sc.pp.log1p(ad)\n",
    "    sc.pp.highly_variable_genes(ad, n_top_genes=int(ad.shape[-1] * 0.1))\n",
    "    \n",
    "    mat = ad[:,ad.var['highly_variable'].values].X.toarray()\n",
    "\n",
    "    scrubber = scr.Scrublet(mat, expected_doublet_rate=0.1, sim_doublet_ratio=3, stdev_doublet_rate=0.025)\n",
    "\n",
    "    scores, predicted = scrubber.scrub_doublets(min_counts=2, min_cells=10, min_gene_variability_pctl=0,\n",
    "                                                n_prin_comps=30, log_transform=False, mean_center=True, \n",
    "                                                synthetic_doublet_umi_subsampling=1., \n",
    "                                                use_approx_neighbors=False,\n",
    "                                                verbose=True\n",
    "                                               )\n",
    "\n",
    "    scrublet_results = pd.DataFrame({'scrublet_score': scores, 'called_doublet': predicted, 'rethresholded_doublets': scores>0.25},\n",
    "                                    index=ad.obs_names\n",
    "                                   )\n",
    "    \n",
    "    scrublet_results.to_csv(outf)\n",
    "    \n",
    "    del ad\n",
    "    del mat\n",
    "    del scrubber"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "scrublet_files = !ls /common/ingn/SCRNA_RAW_DATA/mouse_RTPD/*scrublet.csv\n",
    "\n",
    "ads = []\n",
    "for hf, f in zip(h5adfiles, scrublet_files):\n",
    "    df = pd.read_csv(f, index_col=0, header=0)\n",
    "    \n",
    "    ad = sc.read_h5ad(hf)\n",
    "    ad.obs['scrublet_score'] = df['scrublet_score']\n",
    "    \n",
    "    non_doublets = df.query('rethresholded_doublets==False').index.tolist()\n",
    "    \n",
    "    ad = ad[non_doublets].copy()\n",
    "    ads.append(ad.copy())\n",
    "    \n",
    "    print(f, ad.shape)\n",
    "\n",
    "adata = anndata.concat(ads)\n",
    "adata\n",
    "adata.obs.head()\n",
    "\n",
    "adata.write('RTPD_mouse_scSeq.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad(f\"/Users/gouink/Documents/RTPD1Manuscript/mouse-scSeq/mouse/RTPD_mouse_scSeq.h5ad\")\n",
    "\n",
    "adata.raw = adata\n",
    "\n",
    "sc.pp.highly_variable_genes(adata, n_top_genes=4000, subset=True, flavor='seurat_v3')\n",
    "\n",
    "adata.layers['counts'] = adata[:, adata.var_names].X\n",
    "\n",
    "scvi.model.SCVI.setup_anndata(adata, \n",
    "                              layer= \"counts\")\n",
    "\n",
    "other_params = dict(\n",
    "    dropout_rate= 0.2,\n",
    "    n_latent=32,\n",
    "    n_hidden=256,\n",
    "    n_layers= 2,\n",
    "    gene_likelihood='nb'\n",
    ")\n",
    "\n",
    "plan_params = {\n",
    "                'lr': 1e-3,\n",
    "                'n_epochs_kl_warmup': 2\n",
    "              }\n",
    "\n",
    "trainer_params = {\n",
    "                    'check_val_every_n_epoch': 1,\n",
    "                    'early_stopping': True,\n",
    "                    'early_stopping_patience': 10,\n",
    "                    'early_stopping_monitor': 'reconstruction_loss_validation', \n",
    "                 }\n",
    "\n",
    "vae_ref = scvi.model.SCVI(adata, **other_params)\n",
    "vae_ref.train(max_epochs= 200,\n",
    "              train_size= 0.9,\n",
    "              plan_kwargs= plan_params,\n",
    "              **trainer_params)\n",
    "\n",
    "adata.obsm[\"X_scVI\"] = vae_ref.get_latent_representation()\n",
    "sc.pp.neighbors(adata, use_rep=\"X_scVI\", key_added='scvi')\n",
    "sc.tl.leiden(adata, key_added='leiden', neighbors_key='scvi', resolution=0.6)\n",
    "sc.tl.umap(adata, neighbors_key='scvi')\n",
    "\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"leiden\"],\n",
    "    frameon=False,\n",
    "    legend_loc= 'on data',\n",
    "    ncols=1\n",
    ")\n",
    "\n",
    "adata.write('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi.h5ad')\n",
    "\n",
    "test = anndata.AnnData(adata.raw.X, var=adata.raw.var, obs=adata.obs)\n",
    "test.obsm = adata.obsm\n",
    "\n",
    "sc.pp.normalize_per_cell(test, counts_per_cell_after=10000)\n",
    "sc.pp.log1p(test)\n",
    "\n",
    "model = celltypist.models.Model.load(model = 'Adult_Mouse_Gut.pkl')\n",
    "predictions = celltypist.annotate(test, model = model, majority_voting = True)\n",
    "\n",
    "test.obs['majority_voting'] = predictions.predicted_labels['majority_voting']\n",
    "\n",
    "sc.pl.umap(\n",
    "    test,\n",
    "    color=[\"leiden\", \"majority_voting\"],\n",
    "    frameon=False,\n",
    "    legend_loc= 'on data',\n",
    "    ncols=1\n",
    ")\n",
    "\n",
    "genes = [\n",
    "            'Mki67',\n",
    "            'Epcam',  'Kdr', 'Hbb-bs', 'Vwf',\n",
    "            'Ptprc', \n",
    "            'Ms4a1', 'Cd79a', \n",
    "            'Flt3', 'Irf8', 'H2-Eb1', 'Cd274', 'Itgax', 'Ccr7', 'Ly75', 'Il7r', 'Itgae', 'Kit', 'Havcr2', \n",
    "            'Vim', 'Pdgfrb', 'Col6a1', 'Col4a1', 'Col4a2', 'Acta2', 'Sdc1',\n",
    "            'Cd68', 'Itgam',  'C1qa', 'C1qb', 'Trem2', \n",
    "            'Nkg7', 'Klrd1', 'Klrk1', 'Prf1', 'Ifng', 'Eomes', 'Runx3',\n",
    "            'Cd3e', 'Cd3d', 'Cd4', 'Cd8a', 'Pdcd1', 'Foxp3', 'Ctla4', \n",
    "            'Csf3r', 'S100a8', 'Cd14', \n",
    "            'Ly6d', 'Sell', 'Cd69',  'Ly6c2', 'Ly6g', 'Adgre1', 'Ighg1', 'Tnfrsf17'\n",
    "        ]\n",
    "sc.pl.dotplot(test, var_names=genes, groupby='majority_voting', standard_scale='var')\n",
    "\n",
    "test.obs['majority_voting'].to_csv(\"/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_celltypist.csv\")\n",
    "\n",
    "celltypist_to_broad = {\n",
    "                            \"Cd206 Mac\": \"Myeloid\",\n",
    "                            \"Inflammatory Monocytes\": \"Myeloid\",\n",
    "                            \"CD8+ T cell\": \"T/NK cell\",\n",
    "                            \"DC (CD103+ CD11-b)\": \"Myeloid\",\n",
    "                            \"Resting B cell\": \"B cell\",\n",
    "                            \"DC\": \"Myeloid\",\n",
    "                            \"DC (CD103+ CD11+b)\": \"Myeloid\",\n",
    "                            \"Ccr7 DC\": \"Myeloid\",\n",
    "                            \"Activated CD4+ T cell\": \"T/NK cell\",\n",
    "                            \"Ly6c2 Mono\": \"Myeloid\",\n",
    "                            \"Clec4e mono\": \"Myeloid\",\n",
    "                            \"Neutrophil\": \"Myeloid\",\n",
    "                            \"pDC\": \"Myeloid\",\n",
    "                            \"NK cell\": \"T/NK cell\",\n",
    "                            \"Resting CD4+ T cells\": \"T/NK cell\",\n",
    "                            \"DC (CD103- C2)\": \"Myeloid\",\n",
    "                            \"Monocytes\": \"Myeloid\",\n",
    "                            \"Macrophage\": \"Myeloid\",\n",
    "                            \"Cd11c Mac\": \"Myeloid\",\n",
    "                            \"Fibroblast\": \"toss\",\n",
    "                            \"capillary Aqp7+\": \"toss\",\n",
    "                            \"EarlyGC_1\": \"toss\",\n",
    "                            \"EarlyGC_2\": \"toss\",\n",
    "                            \"Plasma cell\": \"Plasma cell\"\n",
    "                      }\n",
    "\n",
    "test.obs['broad_celltype'] = pd.Categorical(test.obs['majority_voting'].map(celltypist_to_broad))\n",
    "test.obs['broad_celltype'].value_counts()\n",
    "\n",
    "sc.pl.umap(\n",
    "    test,\n",
    "    color=[\"leiden\", \"broad_celltype\"],\n",
    "    frameon=False,\n",
    "    legend_loc= 'on data',\n",
    "    ncols=1\n",
    ")\n",
    "\n",
    "genes = [\n",
    "            'Mki67',\n",
    "            'Epcam',  'Kdr', 'Hbb-bs', 'Vwf',\n",
    "            'Ptprc', \n",
    "            'Ms4a1', 'Cd79a', \n",
    "            'Flt3', 'Irf8', 'H2-Eb1', 'Cd274', 'Itgax', 'Ccr7', 'Ly75', 'Il7r', 'Itgae', 'Kit', 'Havcr2', \n",
    "            'Vim', 'Pdgfrb', 'Col6a1', 'Col4a1', 'Col4a2', 'Acta2', 'Sdc1',\n",
    "            'Cd68', 'Itgam',  'C1qa', 'C1qb', 'Trem2', \n",
    "            'Nkg7', 'Klrd1', 'Klrk1', 'Prf1', 'Ifng', 'Eomes', 'Runx3',\n",
    "            'Cd3e', 'Cd3d', 'Cd4', 'Cd8a', 'Pdcd1', 'Foxp3', 'Ctla4', \n",
    "            'Csf3r', 'S100a8', 'Cd14', \n",
    "            'Ly6d', 'Sell', 'Cd69',  'Ly6c2', 'Ly6g', 'Adgre1', 'Ighg1', 'Tnfrsf17'\n",
    "        ]\n",
    "sc.pl.dotplot(test, var_names=genes, groupby='broad_celltype', standard_scale='var')\n",
    "\n",
    "adata.obs['majority_voting'] = test.obs['majority_voting'].loc[adata.obs.index]\n",
    "adata.obs['broad_celltype'] = test.obs['broad_celltype'].loc[adata.obs.index]\n",
    "adata.write('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi_celltypist.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi_celltypist.h5ad')\n",
    "adata = adata[adata.obs.query(\" broad_celltype != 'toss' \").index].copy()\n",
    "tmp = adata.copy()\n",
    "\n",
    "groupby = 'broad_celltype'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(10,3))\n",
    "\n",
    "_= sns.boxplot(data= counts,\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder)\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Myeloid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi_celltypist.h5ad')\n",
    "# adata = adata[adata.obs.query(\" broad_celltype == 'Myeloid' \").index, :].copy()\n",
    "# adata.raw = adata\n",
    "\n",
    "# sc.pp.highly_variable_genes(adata, n_top_genes=2000, subset=True, flavor='seurat_v3')\n",
    "\n",
    "# adata.layers['counts'] = adata[:, adata.var_names].X\n",
    "\n",
    "# scvi.model.SCVI.setup_anndata(adata, \n",
    "#                               layer= \"counts\")\n",
    "\n",
    "# other_params = dict(\n",
    "#     dropout_rate= 0.2,\n",
    "#     n_latent=32,\n",
    "#     n_hidden=256,\n",
    "#     n_layers= 2,\n",
    "#     gene_likelihood='nb'\n",
    "# )\n",
    "\n",
    "# plan_params = {\n",
    "#                 'lr': 1e-3,\n",
    "#                 'n_epochs_kl_warmup': 2\n",
    "#               }\n",
    "\n",
    "# trainer_params = {\n",
    "#                     'check_val_every_n_epoch': 1,\n",
    "#                     'early_stopping': True,\n",
    "#                     'early_stopping_patience': 10,\n",
    "#                     'early_stopping_monitor': 'reconstruction_loss_validation', \n",
    "#                  }\n",
    "\n",
    "# vae_ref = scvi.model.SCVI(adata, **other_params)\n",
    "# vae_ref.train(max_epochs= 200,\n",
    "#               train_size= 0.9,\n",
    "#               plan_kwargs= plan_params,\n",
    "#               **trainer_params)\n",
    "\n",
    "# adata.obsm[\"X_scVI\"] = vae_ref.get_latent_representation()\n",
    "# sc.pp.neighbors(adata, use_rep=\"X_scVI\", key_added='myeloid_scvi')\n",
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_reclustered.h5ad')\n",
    "sc.tl.leiden(adata, key_added='leiden', neighbors_key='myeloid_scvi', resolution=0.4)\n",
    "# sc.tl.umap(adata, neighbors_key='myeloid_scvi')\n",
    "\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"leiden\", \"majority_voting\"],\n",
    "    legend_loc= 'on data',\n",
    "    frameon=False,\n",
    "    ncols=1\n",
    ")\n",
    "\n",
    "adata.write('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_reclustered.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp = adata.copy()\n",
    "\n",
    "groupby = 'leiden'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(10,3))\n",
    "\n",
    "_= sns.boxplot(data= counts,\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_reclustered.h5ad')\n",
    "sc.pp.normalize_per_cell(adata, counts_per_cell_after=10000)\n",
    "sc.pp.log1p(adata)\n",
    "\n",
    "dg = dgex.leiden_dgex(adata, groupby='leiden')\n",
    "dg.to_csv('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_leiden_dgex.csv')\n",
    "dg = dgex.leiden_dgex(adata, groupby='majority_voting')\n",
    "dg.to_csv('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_majority_voting_dgex.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sc.pl.umap(adata,\n",
    "           color= ['Cd14','Itgax','Mki67','Ly6c2','Adgre1','H2-Ab1','C1qa','Mrc1','Apoe','Trem2','Pltp'],\n",
    "           cmap= 'inferno',\n",
    "           size= 3,\n",
    "           use_raw=False,\n",
    "           show=False,\n",
    "           ncols= 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_reclustered.h5ad')\n",
    "sc.pp.normalize_per_cell(adata, counts_per_cell_after=10000)\n",
    "sc.pp.log1p(adata)\n",
    "\n",
    "GOI = [\"H2-Ab1\", \"C1qa\"]\n",
    "df = pd.DataFrame(adata[:,GOI].X.toarray(), index= adata.obs_names, columns= GOI)\n",
    "df = df.merge(adata.obs, how='left', left_index= True, right_index= True)\n",
    "\n",
    "fig, axd = plt.subplot_mosaic(mosaic= [[\"H2-Ab1\"], ['C1qa']],\n",
    "                              sharex= True,\n",
    "                              sharey= False,\n",
    "                              gridspec_kw= {'hspace': 0.05, 'wspace': 0.05},\n",
    "                              layout= 'constrained',\n",
    "                              figsize= (2,2))\n",
    "\n",
    "for g in GOI:\n",
    "    _= sns.violinplot(data= df,\n",
    "                        x= 'majority_voting',\n",
    "                        y= g,\n",
    "                        hue= 'treatment',\n",
    "                        order= ['Cd206 Mac', 'Ly6c2 Mono'],\n",
    "                        hue_order= ['IgG', 'PD1', 'RTIgG', 'RTPD1'],\n",
    "                        linewidth= 0.5,\n",
    "                        ax= axd[g])\n",
    "    _= axd[g].set_xlabel('')\n",
    "    _= axd[g].set_xticklabels(['Cd206\\nMac', 'Inflammatory\\nMonocytes'], fontsize= 8)\n",
    "    _= axd[g].set_ylabel('log norm\\nexpression', fontsize= 8)\n",
    "    _= axd[g].set_title(g, fontsize= 8)\n",
    "    _= axd[g].set_yticks(np.arange(start= 0, stop= 10, step= 2))\n",
    "    _= axd[g].set_yticklabels([str(x) for x in np.arange(start= 0, stop= 10, step= 2)], fontsize= 8)\n",
    "    _= axd[g].grid(visible= False)\n",
    "    _= axd[g].get_legend().remove()\n",
    "\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_myeloid_violin.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp = adata.copy()\n",
    "\n",
    "groupby = 'majority_voting'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(10,3))\n",
    "\n",
    "_= sns.boxplot(data= counts,\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder)\n",
    "_= axs.set_xticklabels(order, rotation=90)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "counts.query(\" treatment == 'IgG' & majority_voting == 'Cd206 Mac' \")['percent'].median()\n",
    "counts.query(\" treatment == 'RTIgG' & majority_voting == 'Cd206 Mac' \")['percent'].median()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "57.69736842105263 / 33.500781766545956"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi_celltypist.h5ad')\n",
    "test = anndata.AnnData(adata.raw.X, var=adata.raw.var, obs=adata.obs)\n",
    "\n",
    "adata = test[test.obs.query(\" broad_celltype == 'Myeloid' \").index].copy()\n",
    "sc.pp.filter_genes(adata, min_cells=0.1*(len(adata)))\n",
    "print(adata.shape)\n",
    "\n",
    "tmp = pd.DataFrame(index= adata.var_names, columns=['mouse_name','human_name'])\n",
    "tmp['mouse_name'] = adata.var_names\n",
    "\n",
    "orth = pd.read_csv('/Users/gouink/Documents/GeneLists/Ensembl_Human_Mouse_Ortholog.txt',index_col=None,header=0)\n",
    "for i in tmp.index.tolist():\n",
    "    match = orth.query(\" Mousegenename == @i \")['Genename'].tolist()\n",
    "    if len(match)==0:\n",
    "        continue\n",
    "    elif len(match)>1:\n",
    "        match = match[0]\n",
    "        tmp.loc[i,'human_name'] = match\n",
    "    else:\n",
    "        tmp.loc[i,'human_name'] = match[0]\n",
    "\n",
    "tmp.dropna(how='any',inplace=True)\n",
    "\n",
    "all_mouse_genes = tmp['mouse_name'].tolist()\n",
    "all_human_genes = tmp['human_name'].tolist()\n",
    "\n",
    "mouse_dgex = pd.read_csv('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_leiden_dgex.csv', index_col=0, header=0)\n",
    "human_dgex = pd.read_csv('/Users/gouink/Documents/RTPD1Manuscript/Human/manuscript_review_analysis/analysis/myeloid_filtered/leiden_dgex.csv', index_col=0, header=0)\n",
    "\n",
    "olap_df = pd.DataFrame(index= [f'mouse_{x}' for x in sorted(mouse_dgex['group'].unique().tolist())],\n",
    "                       columns= [f'human_{x}' for x in sorted(human_dgex['group'].unique().tolist())])\n",
    "\n",
    "for i in olap_df.index:\n",
    "\n",
    "    g = int(i.split('_')[-1])\n",
    "    mouse_genes = mouse_dgex.query(\" group == @g & pvals_adj < 0.005 & logfoldchanges >= 1.0 & pct_nz_group >= 0.3 \")['names'].tolist()\n",
    "    mouse_genes = list(set(mouse_genes) & set(all_mouse_genes))\n",
    "    mouse_ortholog_genes = [tmp.loc[x,'human_name'] for x in mouse_genes]\n",
    "    num_mouse = len(mouse_ortholog_genes)\n",
    "\n",
    "    for j in olap_df.columns:\n",
    "        \n",
    "        g = int(j.split('_')[-1])\n",
    "        human_genes = human_dgex.query(\" group == @g & pvals_adj < 0.005 & logfoldchanges >= 1.0 & pct_nz_group >= 0.3 \")['names'].tolist()\n",
    "        human_genes = list(set(human_genes) & set(all_human_genes))\n",
    "        num_human = len(human_genes)\n",
    "\n",
    "        overlap = len(set(mouse_ortholog_genes) & set(human_genes))\n",
    "        pval = scipy.stats.hypergeom.sf(overlap-1, len(all_mouse_genes), num_mouse, num_human)\n",
    "\n",
    "        olap_df.loc[i,j] = [num_mouse, num_human, overlap, pval]\n",
    "        # print(f\"{i}<->{j}: Mouse: {num_mouse}, Human: {num_human}, Overlap: {overlap}, pval: {pval}\")\n",
    "\n",
    "olap_df\n",
    "\n",
    "olap_size = olap_df.applymap(lambda x: x[2]/x[0] if x[0]!=0 else 0)\n",
    "pval = olap_df.applymap(lambda x: -1*np.log10(x[3]))\n",
    "\n",
    "gf.heatmap2(data= olap_size,\n",
    "            vmin= 0,\n",
    "            vmax= olap_size.max().max(),\n",
    "            cellsize= pval,\n",
    "            cmap= 'viridis',\n",
    "            annot= pval,\n",
    "            annot_kws= {'fontsize': 8, 'weight': 'normal'},\n",
    "            ref_sizes= [1e-2, 1e-4, 1e-6],\n",
    "            ref_labels= ['1e-2','1e-4','1e-6'],\n",
    "            figsize= (8,8),\n",
    "            fontsize= 12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi_celltypist.h5ad')\n",
    "test = anndata.AnnData(adata.raw.X, var=adata.raw.var, obs=adata.obs)\n",
    "\n",
    "adata = test[test.obs.query(\" broad_celltype == 'Myeloid' \").index].copy()\n",
    "sc.pp.filter_genes(adata, min_cells=0.1*(len(adata)))\n",
    "print(adata.shape)\n",
    "\n",
    "tmp = pd.DataFrame(index= adata.var_names, columns=['mouse_name','human_name'])\n",
    "tmp['mouse_name'] = adata.var_names\n",
    "\n",
    "orth = pd.read_csv('/Users/gouink/Documents/GeneLists/Ensembl_Human_Mouse_Ortholog.txt',index_col=None,header=0)\n",
    "for i in tmp.index.tolist():\n",
    "    match = orth.query(\" Mousegenename == @i \")['Genename'].tolist()\n",
    "    if len(match)==0:\n",
    "        continue\n",
    "    elif len(match)>1:\n",
    "        match = match[0]\n",
    "        tmp.loc[i,'human_name'] = match\n",
    "    else:\n",
    "        tmp.loc[i,'human_name'] = match[0]\n",
    "\n",
    "tmp.dropna(how='any',inplace=True)\n",
    "\n",
    "all_mouse_genes = tmp['mouse_name'].tolist()\n",
    "all_human_genes = tmp['human_name'].tolist()\n",
    "\n",
    "mouse_dgex = pd.read_csv('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_majority_voting_dgex.csv', index_col=0, header=0)\n",
    "human_dgex = pd.read_csv('/Users/gouink/Documents/RTPD1Manuscript/Human/manuscript_review_analysis/analysis/myeloid_filtered/leiden_dgex.csv', index_col=0, header=0)\n",
    "\n",
    "olap_df = pd.DataFrame(index= [f'mouse_{x}' for x in sorted(mouse_dgex['group'].unique().tolist())],\n",
    "                       columns= [f'human_{x}' for x in sorted(human_dgex['group'].unique().tolist())])\n",
    "\n",
    "for i in olap_df.index:\n",
    "\n",
    "    g = i.split('_')[-1]\n",
    "    mouse_genes = mouse_dgex.query(\" group == @g & pvals_adj < 0.005 & logfoldchanges >= 1.0 & pct_nz_group >= 0.3 \")['names'].tolist()\n",
    "    mouse_genes = list(set(mouse_genes) & set(all_mouse_genes))\n",
    "    mouse_ortholog_genes = [tmp.loc[x,'human_name'] for x in mouse_genes]\n",
    "    num_mouse = len(mouse_ortholog_genes)\n",
    "\n",
    "    for j in olap_df.columns:\n",
    "        \n",
    "        g = int(j.split('_')[-1])\n",
    "        human_genes = human_dgex.query(\" group == @g & pvals_adj < 0.005 & logfoldchanges >= 1.0 & pct_nz_group >= 0.3 \")['names'].tolist()\n",
    "        human_genes = list(set(human_genes) & set(all_human_genes))\n",
    "        num_human = len(human_genes)\n",
    "\n",
    "        overlap = len(set(mouse_ortholog_genes) & set(human_genes))\n",
    "        pval = scipy.stats.hypergeom.sf(overlap-1, len(all_mouse_genes), num_mouse, num_human)\n",
    "\n",
    "        olap_df.loc[i,j] = [num_mouse, num_human, overlap, pval]\n",
    "        # print(f\"{i}<->{j}: Mouse: {num_mouse}, Human: {num_human}, Overlap: {overlap}, pval: {pval}\")\n",
    "\n",
    "olap_df\n",
    "\n",
    "olap_size = olap_df.applymap(lambda x: x[2]/x[0] if x[0]!=0 else 0)\n",
    "pval = olap_df.applymap(lambda x: -1*np.log10(x[3]))\n",
    "\n",
    "gf.heatmap2(data= olap_size,\n",
    "            vmin= 0,\n",
    "            vmax= olap_size.max().max(),\n",
    "            cellsize= pval,\n",
    "            cmap= 'viridis',\n",
    "            annot= pval,\n",
    "            annot_kws= {'fontsize': 8, 'weight': 'normal'},\n",
    "            ref_sizes= [1e-2, 1e-4, 1e-6],\n",
    "            ref_labels= ['1e-2','1e-4','1e-6'],\n",
    "            figsize= (8,8),\n",
    "            fontsize= 12)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## T-cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi_celltypist.h5ad')\n",
    "adata = adata[adata.obs.query(\" broad_celltype == 'T/NK cell' \").index, :].copy()\n",
    "adata.raw = adata\n",
    "\n",
    "sc.pp.highly_variable_genes(adata, n_top_genes=2000, subset=True, flavor='seurat_v3')\n",
    "\n",
    "adata.layers['counts'] = adata[:, adata.var_names].X\n",
    "\n",
    "scvi.model.SCVI.setup_anndata(adata, \n",
    "                              layer= \"counts\")\n",
    "\n",
    "other_params = dict(\n",
    "    dropout_rate= 0.2,\n",
    "    n_latent=32,\n",
    "    n_hidden=256,\n",
    "    n_layers= 2,\n",
    "    gene_likelihood='nb'\n",
    ")\n",
    "\n",
    "plan_params = {\n",
    "                'lr': 1e-3,\n",
    "                'n_epochs_kl_warmup': 2\n",
    "              }\n",
    "\n",
    "trainer_params = {\n",
    "                    'check_val_every_n_epoch': 1,\n",
    "                    'early_stopping': True,\n",
    "                    'early_stopping_patience': 10,\n",
    "                    'early_stopping_monitor': 'reconstruction_loss_validation', \n",
    "                 }\n",
    "\n",
    "vae_ref = scvi.model.SCVI(adata, **other_params)\n",
    "vae_ref.train(max_epochs= 200,\n",
    "              train_size= 0.9,\n",
    "              plan_kwargs= plan_params,\n",
    "              **trainer_params)\n",
    "\n",
    "adata.obsm[\"X_scVI\"] = vae_ref.get_latent_representation()\n",
    "sc.pp.neighbors(adata, use_rep=\"X_scVI\", key_added='tcell_scvi')\n",
    "sc.tl.leiden(adata, key_added='leiden', neighbors_key='tcell_scvi', resolution=0.3)\n",
    "sc.tl.umap(adata, neighbors_key='tcell_scvi')\n",
    "\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"leiden\", \"majority_voting\"],\n",
    "    frameon=False,\n",
    "    ncols=1\n",
    ")\n",
    "\n",
    "adata.write('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "test = anndata.AnnData(adata.raw.X, var=adata.raw.var, obs=adata.obs)\n",
    "test.obsm = adata.obsm\n",
    "\n",
    "sc.pp.normalize_per_cell(test, counts_per_cell_after=10000)\n",
    "sc.pp.log1p(test)\n",
    "\n",
    "_= sc.pl.umap(test,\n",
    "                color= ['Cd3e','Cd4','Cd8a','Ncr1','Gzma','Mki67','Foxp3','Ctla4','Tnfrsf4','Il10','Pdcd1','Lag3','Tcf7','Prf1','Gzmb'],\n",
    "                cmap= 'inferno',\n",
    "                size= 20,\n",
    "                use_raw=False,\n",
    "                show=False,\n",
    "                ncols= 4)\n",
    "\n",
    "test2 = test[test.obs.query(\" majority_voting == 'CD8+ T cell' \").index, :].copy()\n",
    "rm_clust = test2.obs['leiden'].value_counts().index[test2.obs['leiden'].value_counts() < 2].tolist()\n",
    "test2 = test2[test2.obs.query(\" leiden not in @rm_clust \").index, :].copy()\n",
    "# dg = dgex.leiden_dgex(test2, groupby='leiden')\n",
    "# dg.to_csv('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/CD8T_leiden_dgex.csv')\n",
    "_= sc.pl.violin(test2,\n",
    "                keys= ['Cd3e','Cd4','Cd8a','Mki67','Pdcd1','Lag3','Tcf7','Prf1','Gzmb'],\n",
    "                groupby= 'leiden',\n",
    "                order= ['1','2','3','4'],\n",
    "                use_raw= False,\n",
    "                show= False)\n",
    "\n",
    "test2 = test[test.obs.query(\" majority_voting == 'Resting CD4+ T cells' | majority_voting == 'Activated CD4+ T cell' \").index, :].copy()\n",
    "rm_clust = test2.obs['leiden'].value_counts().index[test2.obs['leiden'].value_counts() < 2].tolist()\n",
    "test2 = test2[test2.obs.query(\" leiden not in @rm_clust \").index, :].copy()\n",
    "# dg = dgex.leiden_dgex(test2, groupby='leiden')\n",
    "# dg.to_csv('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/CD4T_leiden_dgex.csv')\n",
    "_= sc.pl.violin(test2,\n",
    "                keys= ['Cd3e','Cd4','Cd8a','Mki67','Foxp3','Ctla4','Tnfrsf4','Il10','Pdcd1','Tcf7','Gata3','Il2ra','Il2rb'],\n",
    "                groupby= 'leiden',\n",
    "                order= ['0','1','3','4'],\n",
    "                use_raw= False,\n",
    "                show= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp = adata.copy()\n",
    "\n",
    "groupby = 'leiden'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(10,3))\n",
    "\n",
    "_= sns.boxplot(data= counts,\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder)\n",
    "_= axs.set_title(\"Normalized to total T-cells\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp = adata.copy()\n",
    "\n",
    "groupby = 'majority_voting'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(10,3))\n",
    "\n",
    "_= sns.boxplot(data= counts,\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder)\n",
    "_= axs.set_xticklabels(order, rotation=90)\n",
    "_= axs.set_title(\"Normalized to total T-cells\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "tmp = adata[adata.obs.query(\" majority_voting == 'CD8+ T cell' \").index, :].copy()\n",
    "\n",
    "groupby = 'leiden'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(10,3))\n",
    "\n",
    "_= sns.boxplot(data= counts,\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder)\n",
    "\n",
    "_= axs.set_title(\"Normalized to total CD8T\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "tmp = adata[adata.obs.query(\" majority_voting == 'Resting CD4+ T cells' | majority_voting == 'Activated CD4+ T cell' \").index, :].copy()\n",
    "\n",
    "groupby = 'leiden'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(10,3))\n",
    "\n",
    "_= sns.boxplot(data= counts,\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder)\n",
    "\n",
    "_= axs.set_title(\"Normalized to total CD4T\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final figures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi_celltypist.h5ad')\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(2,2))\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"leiden\"],\n",
    "    frameon=False,\n",
    "    legend_loc= 'on data',\n",
    "    ncols=1,\n",
    "    show= False,\n",
    "    ax= axs\n",
    ")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_global_umap_leiden.pdf', bbox_inches='tight')\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(2,2))\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"majority_voting\"],\n",
    "    frameon=False,\n",
    "    legend_loc= 'on data',\n",
    "    legend_fontsize= 2,\n",
    "    ncols=1,\n",
    "    show= False,\n",
    "    ax= axs\n",
    ")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_global_umap_celltypist.pdf', bbox_inches='tight')\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(2,2))\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"broad_celltype\"],\n",
    "    frameon=False,\n",
    "    groups= ['B cell', 'Myeloid', 'T/NK cell', 'Plasma cell'],\n",
    "    legend_loc= 'on data',\n",
    "    ncols=1,\n",
    "    show= False,\n",
    "    ax= axs\n",
    ")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_global_umap_celltypistBroad.pdf', bbox_inches='tight')\n",
    "\n",
    "\n",
    "# test = anndata.AnnData(adata.raw.X, var=adata.raw.var, obs=adata.obs)\n",
    "# test.obsm = adata.obsm\n",
    "\n",
    "# sc.pp.normalize_per_cell(test, counts_per_cell_after=10000)\n",
    "# sc.pp.log1p(test)\n",
    "\n",
    "# genes = [\n",
    "#             'Mki67',\n",
    "#             'Epcam',  'Kdr', 'Hbb-bs', 'Vwf',\n",
    "#             'Ptprc', \n",
    "#             'Ms4a1', 'Cd79a', \n",
    "#             'Flt3', 'Irf8', 'H2-Eb1', 'Cd274', 'Itgax', 'Ccr7', 'Ly75', 'Il7r', 'Itgae', 'Kit', 'Havcr2', \n",
    "#             'Vim', 'Pdgfrb', 'Col6a1', 'Col4a1', 'Col4a2', 'Acta2', 'Sdc1',\n",
    "#             'Cd68', 'Itgam',  'C1qa', 'C1qb', 'Trem2', \n",
    "#             'Nkg7', 'Klrd1', 'Klrk1', 'Prf1', 'Ifng', 'Eomes', 'Runx3',\n",
    "#             'Cd3e', 'Cd3d', 'Cd4', 'Cd8a', 'Pdcd1', 'Foxp3', 'Ctla4', \n",
    "#             'Csf3r', 'S100a8', 'Cd14', \n",
    "#             'Ly6d', 'Sell', 'Cd69',  'Ly6c2', 'Ly6g', 'Adgre1', 'Ighg1', 'Tnfrsf17'\n",
    "#         ]\n",
    "\n",
    "# fig,axs = plt.subplots(figsize=(4,4))\n",
    "# sc.pl.dotplot(test, \n",
    "#               var_names=genes,\n",
    "#               use_raw= False,\n",
    "#               groupby='broad_celltype', \n",
    "#               standard_scale='var', \n",
    "#               dendrogram= True,\n",
    "#               show= False,\n",
    "#               ax= axs)\n",
    "# fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/timecourseRT_v4/figures/timecourse_global_expr_dotplot.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_reclustered.h5ad')\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(2,2))\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"leiden\"],\n",
    "    frameon=False,\n",
    "    legend_loc= 'right margin',\n",
    "    legend_fontsize= 8,\n",
    "    ncols=1,\n",
    "    show= False,\n",
    "    ax= axs\n",
    ")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_myeloid_umap_leiden.pdf', bbox_inches='tight')\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(2,2))\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"majority_voting\"],\n",
    "    frameon=False,\n",
    "    legend_loc= 'right margin',\n",
    "    legend_fontsize= 8,\n",
    "    ncols=1,\n",
    "    show= False,\n",
    "    ax= axs\n",
    ")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_myeloid_umap_celltypist.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_reclustered.h5ad')\n",
    "tmp = adata.copy()\n",
    "\n",
    "groupby = 'majority_voting'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "# order = sorted(adata.obs[groupby].unique().tolist())\n",
    "order = ['Cd206 Mac', 'Inflammatory Monocytes']\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(2,3))\n",
    "\n",
    "_= sns.boxplot(data= counts.reset_index(),\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder,\n",
    "                   fliersize= 0,\n",
    "                   linewidth= 0.5)\n",
    "_= sns.stripplot(data= counts.reset_index(),\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder,\n",
    "                   dodge= True,\n",
    "                   s= 3,\n",
    "                   legend= None)\n",
    "_= axs.set_xticklabels(order, rotation=90)\n",
    "_= axs.set_ylim(0, 70)\n",
    "_= axs.get_legend().remove()\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_myeloid_boxplot_celltypist.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_reclustered.h5ad')\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1',\n",
    "            'B3': 'PD1',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1',\n",
    "            'B6': 'RTPD1',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1',\n",
    "            'mC1': 'PD1',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1',\n",
    "            'mF1': 'RTPD1'\n",
    "        }\n",
    "\n",
    "adata.obs['treatment'] = adata.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = adata.obs.groupby(by=['treatment', 'majority_voting']).count()['sampleID']\n",
    "\n",
    "t1 = 'IgG'\n",
    "t2 = 'PD1'\n",
    "\n",
    "total_per_cluster = counts[t1, :] + counts[t2, :]\n",
    "total = total_per_cluster.sum()\n",
    "\n",
    "f_exp_IgG = (total_per_cluster / total) * counts[t1,:].sum()\n",
    "f_exp_RTIgG = (total_per_cluster / total) * counts[t2,:].sum()\n",
    "\n",
    "result = scipy.stats.chisquare(np.array([counts[t1, :], counts[t2, :]]).T, \n",
    "                                np.array([f_exp_IgG, f_exp_RTIgG]).T,\n",
    "                                axis= 1)\n",
    "\n",
    "result = pd.DataFrame(result.pvalue, index=counts[t1, :].index)\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "counts['IgG', :].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "counts['RTIgG', :].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/combined_RTPDv4_scvi_celltypist.h5ad')\n",
    "test = anndata.AnnData(adata.raw.X, var=adata.raw.var, obs=adata.obs)\n",
    "\n",
    "adata = test[test.obs.query(\" broad_celltype == 'Myeloid' \").index].copy()\n",
    "sc.pp.filter_genes(adata, min_cells=0.1*(len(adata)))\n",
    "print(adata.shape)\n",
    "\n",
    "tmp = pd.DataFrame(index= adata.var_names, columns=['mouse_name','human_name'])\n",
    "tmp['mouse_name'] = adata.var_names\n",
    "\n",
    "orth = pd.read_csv('/Users/gouink/Documents/GeneLists/Ensembl_Human_Mouse_Ortholog.txt',index_col=None,header=0)\n",
    "for i in tmp.index.tolist():\n",
    "    match = orth.query(\" Mousegenename == @i \")['Genename'].tolist()\n",
    "    if len(match)==0:\n",
    "        continue\n",
    "    elif len(match)>1:\n",
    "        match = match[0]\n",
    "        tmp.loc[i,'human_name'] = match\n",
    "    else:\n",
    "        tmp.loc[i,'human_name'] = match[0]\n",
    "\n",
    "tmp.dropna(how='any',inplace=True)\n",
    "\n",
    "all_mouse_genes = tmp['mouse_name'].tolist()\n",
    "all_human_genes = tmp['human_name'].tolist()\n",
    "\n",
    "mouse_dgex = pd.read_csv('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/myeloid_majority_voting_dgex.csv', index_col=0, header=0)\n",
    "human_dgex = pd.read_csv('/Users/gouink/Documents/RTPD1Manuscript/Human/manuscript_review_analysis/analysis/myeloid_filtered/leiden_dgex.csv', index_col=0, header=0)\n",
    "\n",
    "olap_df = pd.DataFrame(index= [f'mouse_{x}' for x in sorted(mouse_dgex['group'].unique().tolist())],\n",
    "                       columns= [f'human_{x}' for x in sorted(human_dgex['group'].unique().tolist())])\n",
    "\n",
    "for i in olap_df.index:\n",
    "\n",
    "    g = i.split('_')[-1]\n",
    "    mouse_genes = mouse_dgex.query(\" group == @g & pvals_adj < 0.005 & logfoldchanges >= 1.0 & pct_nz_group >= 0.3 \")['names'].tolist()\n",
    "    mouse_genes = list(set(mouse_genes) & set(all_mouse_genes))\n",
    "    mouse_ortholog_genes = [tmp.loc[x,'human_name'] for x in mouse_genes]\n",
    "    num_mouse = len(mouse_ortholog_genes)\n",
    "\n",
    "    for j in olap_df.columns:\n",
    "        \n",
    "        g = int(j.split('_')[-1])\n",
    "        human_genes = human_dgex.query(\" group == @g & pvals_adj < 0.005 & logfoldchanges >= 1.0 & pct_nz_group >= 0.3 \")['names'].tolist()\n",
    "        human_genes = list(set(human_genes) & set(all_human_genes))\n",
    "        num_human = len(human_genes)\n",
    "\n",
    "        overlap = len(set(mouse_ortholog_genes) & set(human_genes))\n",
    "        pval = scipy.stats.hypergeom.sf(overlap-1, len(all_mouse_genes), num_mouse, num_human)\n",
    "\n",
    "        olap_df.loc[i,j] = [num_mouse, num_human, overlap, pval]\n",
    "        # print(f\"{i}<->{j}: Mouse: {num_mouse}, Human: {num_human}, Overlap: {overlap}, pval: {pval}\")\n",
    "\n",
    "olap_df\n",
    "\n",
    "olap_size = olap_df.applymap(lambda x: x[2]/x[0] if x[0]!=0 else 0)\n",
    "pval = olap_df.applymap(lambda x: -1*np.log10(x[3]))\n",
    "pval['reference'] = np.arange(len(pval))\n",
    "olap_size['reference'] = np.ones(len(pval))\n",
    "\n",
    "g = gf.heatmap2(data= olap_size,\n",
    "                vmin= 0,\n",
    "                vmax= olap_size.max().max(),\n",
    "                cellsize= pval,\n",
    "                cmap= 'viridis',\n",
    "                annot= pval,\n",
    "                annot_kws= {'fontsize': 8, 'weight': 'normal'},\n",
    "                ref_sizes= [10, 20 ,30],\n",
    "                ref_labels= ['1e-2','1e-4','1e-6'],\n",
    "                figsize= (8,8),\n",
    "                fontsize= 12)\n",
    "\n",
    "g.figure.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_myeloid_celltypist_geneintersection.pdf', bbox_inches='tight', transparent= True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(2,2))\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"leiden\"],\n",
    "    frameon=False,\n",
    "    legend_loc= 'right margin',\n",
    "    legend_fontsize= 8,\n",
    "    ncols=1,\n",
    "    show= False,\n",
    "    ax= axs\n",
    ")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_tcell_umap_leiden.pdf', bbox_inches='tight')\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(2,2))\n",
    "sc.pl.umap(\n",
    "    adata,\n",
    "    color=[\"majority_voting\"],\n",
    "    frameon=False,\n",
    "    legend_loc= 'right margin',\n",
    "    legend_fontsize= 8,\n",
    "    ncols=1,\n",
    "    show= False,\n",
    "    ax= axs\n",
    ")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_tcell_umap_celltypist.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "tmp = adata.copy()\n",
    "\n",
    "groupby = 'leiden'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(10,3))\n",
    "\n",
    "_= sns.boxplot(data= counts,\n",
    "                x= groupby,\n",
    "                y= 'percent',\n",
    "                hue= 'treatment',\n",
    "                order= order,\n",
    "                hue_order= horder)\n",
    "_= axs.set_title(\"Normalized to total T-cells\")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_tcell_leiden_boxplot.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "tmp = adata[adata.obs.query(\" majority_voting == 'CD8+ T cell' \").index, :].copy()\n",
    "\n",
    "groupby = 'leiden'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(10,3))\n",
    "\n",
    "_= sns.boxplot(data= counts,\n",
    "                x= groupby,\n",
    "                y= 'percent',\n",
    "                hue= 'treatment',\n",
    "                order= order,\n",
    "                hue_order= horder)\n",
    "\n",
    "_= axs.set_title(\"Normalized to total CD8T\")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_CD8T_leiden_boxplot.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "\n",
    "tmp = adata.copy()\n",
    "\n",
    "groupby = 'majority_voting'\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1_nonresp',\n",
    "            'B3': 'PD1_resp',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1_nonresp',\n",
    "            'B6': 'RTPD1_resp',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1_nonresp',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1_nonresp',\n",
    "            'mC1': 'PD1_resp',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1_nonresp',\n",
    "            'mF1': 'RTPD1_resp'\n",
    "        }\n",
    "\n",
    "tmp.obs['treatment'] = tmp.obs['treatment'].astype(str)\n",
    "tmp.obs['treatment_response'] = tmp.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = pf.pct_df_faster(tmp,\n",
    "                          groupby= groupby,\n",
    "                          rep= 'sampleID',\n",
    "                          xcat= 'treatment')\n",
    "counts\n",
    "\n",
    "order = sorted(adata.obs[groupby].unique().tolist())\n",
    "order = ['Resting CD4+ T cells', 'Activated CD4+ T cell']\n",
    "horder = ['IgG', 'PD1', 'RTIgG', 'RTPD1']\n",
    "\n",
    "fig,axs = plt.subplots(figsize=(2,3))\n",
    "\n",
    "_= sns.boxplot(data= counts.reset_index(),\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   hue_order= horder,\n",
    "                   fliersize= 0,\n",
    "                   linewidth= 0.5)\n",
    "\n",
    "_= sns.stripplot(data= counts.reset_index(),\n",
    "                   x= groupby,\n",
    "                   y= 'percent',\n",
    "                   hue= 'treatment',\n",
    "                   order= order,\n",
    "                   s= 3,\n",
    "                   dodge= True,\n",
    "                   hue_order= horder,\n",
    "                   legend= None)\n",
    "\n",
    "_= axs.set_xticklabels(order, rotation=90)\n",
    "_= axs.set_ylim(0, 70)\n",
    "_= axs.get_legend().remove()\n",
    "_= axs.set_title(\"Normalized to total T-cells\")\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_tcell_celltypist_boxplot.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# adata_orig = sc.read_h5ad(f\"/Users/gouink/Documents/RTPD1Manuscript/mouse-scSeq/mouse/RTPD_mouse_scSeq.h5ad\")\n",
    "# sc.pp.normalize_per_cell(adata_orig, counts_per_cell_after=10000)\n",
    "# sc.pp.log1p(adata_orig)\n",
    "\n",
    "# sigs = ['CD8Activation', 'G1', 'G2', 'Cytolytics', 'TerminalDiff']\n",
    "\n",
    "# for sig in sigs:\n",
    "#     GOI = pd.read_csv(f'/Users/gouink/Documents/GeneLists/Mouse_{sig}.csv', index_col=0, header=0)\n",
    "#     GOI = GOI.iloc[:, 1].tolist()\n",
    "#     sc.tl.score_genes(adata_orig,\n",
    "#                       gene_list= GOI,\n",
    "#                       score_name= sig,\n",
    "#                       use_raw= False)\n",
    "    \n",
    "# adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "# adata = adata[adata.obs.query(\" majority_voting == 'CD8+ T cell' \").index, :].copy()\n",
    "# adata.obs[sigs] = adata_orig.obs.loc[adata.obs.index, sigs]\n",
    "\n",
    "# tx_map = {\n",
    "#             'B1': 'IgG',\n",
    "#             'B2': 'PD1',\n",
    "#             'B3': 'PD1',\n",
    "#             'B4': 'RTIgG',\n",
    "#             'B5': 'RTPD1',\n",
    "#             'B6': 'RTPD1',\n",
    "#             'C4': 'RTIgG',\n",
    "#             'C5': 'RTPD1',\n",
    "#             'mA1': 'IgG',\n",
    "#             'mB1': 'PD1',\n",
    "#             'mC1': 'PD1',\n",
    "#             'mD1': 'RTIgG',\n",
    "#             'mE1': 'RTPD1',\n",
    "#             'mF1': 'RTPD1'\n",
    "#         }\n",
    "\n",
    "# df = adata.obs.groupby(by=['sampleID']).median(numeric_only=True)\n",
    "# df['treatment'] = [tx_map[x] for x in df.index]\n",
    "\n",
    "fig,axs = plt.subplots(nrows= 2,\n",
    "                       ncols= 3,\n",
    "                       sharex= True,\n",
    "                       sharey= False,\n",
    "                       gridspec_kw={'hspace':0.3, 'wspace':0.5},\n",
    "                       figsize=(6,3))\n",
    "\n",
    "for sig, ax in zip(sigs, axs.flat):\n",
    "\n",
    "    _= sns.boxplot(data= df,\n",
    "                    x= 'treatment',\n",
    "                    y= sig,\n",
    "                    fliersize= 0,\n",
    "                    order= ['IgG', 'PD1', 'RTIgG', 'RTPD1'],\n",
    "                    linewidth= 0.5,\n",
    "                    ax= ax)\n",
    "\n",
    "    _= sns.stripplot(data= df,\n",
    "                    x= 'treatment',\n",
    "                    y= sig,\n",
    "                    s= 3,\n",
    "                    order= ['IgG', 'PD1', 'RTIgG', 'RTPD1'],\n",
    "                    ax= ax)\n",
    "    \n",
    "    _= ax.set_xticklabels(['IgG', 'PD1', 'RTIgG', 'RTPD1'], fontsize=8, rotation=45)\n",
    "    _= ax.set_title(f\"{sig}\", fontsize=8)\n",
    "    _= ax.set_xlabel('')\n",
    "    _= ax.set_ylabel('')\n",
    "    # _= ax.set_ylim((0, df[sig].max()))\n",
    "                   \n",
    "_= fig.suptitle('CD8T Signatures')\n",
    "fig.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_CD8T_signatures.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata_orig = sc.read_h5ad(f\"/Users/gouink/Documents/RTPD1Manuscript/mouse-scSeq/mouse/RTPD_mouse_scSeq.h5ad\")\n",
    "sc.pp.normalize_per_cell(adata_orig, counts_per_cell_after=10000)\n",
    "sc.pp.log1p(adata_orig)\n",
    "\n",
    "sigs = ['CD8Activation', 'G1', 'G2', 'Cytolytics', 'TerminalDiff']\n",
    "\n",
    "for sig in sigs:\n",
    "    GOI = pd.read_csv(f'/Users/gouink/Documents/GeneLists/Mouse_{sig}.csv', index_col=0, header=0)\n",
    "    GOI = GOI.iloc[:, 1].tolist()\n",
    "    sc.tl.score_genes(adata_orig,\n",
    "                      gene_list= GOI,\n",
    "                      score_name= sig,\n",
    "                      use_raw= False)\n",
    "    \n",
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "adata = adata[adata.obs.query(\" majority_voting == 'CD8+ T cell' \").index, :].copy()\n",
    "adata.obs[sigs] = adata_orig.obs.loc[adata.obs.index, sigs]\n",
    "\n",
    "# cellBC = []\n",
    "# for s in adata.obs['sampleID'].unique():\n",
    "#     tmp = adata.obs.query(\" sampleID == @s \").copy()\n",
    "#     if len(tmp) <= 100:\n",
    "#         cellBC += tmp.index.tolist()\n",
    "#     else:\n",
    "#         cellBC += tmp.sample(100).index.tolist()\n",
    "\n",
    "# adata = adata[cellBC]\n",
    "# adata.obs['sampleID'].value_counts()\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1',\n",
    "            'B3': 'PD1',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1',\n",
    "            'B6': 'RTPD1',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1',\n",
    "            'mC1': 'PD1',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1',\n",
    "            'mF1': 'RTPD1'\n",
    "        }\n",
    "\n",
    "adata.obs['treatment'] = adata.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "sc.pl.violin(adata,\n",
    "             groupby= 'treatment',\n",
    "             keys= sigs,\n",
    "             stripplot= False,\n",
    "             show= False)\n",
    "plt.savefig('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/figures/RTPD_CD8T_signatures_violinplot.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "x1 = adata.obs.query(\" treatment == 'IgG' \")['Cytolytics'].to_numpy()\n",
    "x2 = adata.obs.query(\" treatment == 'PD1' \")['Cytolytics'].to_numpy()\n",
    "\n",
    "scipy.stats.ranksums(x1, x2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(x1)\n",
    "len(x2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = sc.read_h5ad('/Users/gouink/Documents/RTPD1Manuscript/RTPD_v4/tcell_reclustered.h5ad')\n",
    "\n",
    "tx_map = {\n",
    "            'B1': 'IgG',\n",
    "            'B2': 'PD1',\n",
    "            'B3': 'PD1',\n",
    "            'B4': 'RTIgG',\n",
    "            'B5': 'RTPD1',\n",
    "            'B6': 'RTPD1',\n",
    "            'C4': 'RTIgG',\n",
    "            'C5': 'RTPD1',\n",
    "            'mA1': 'IgG',\n",
    "            'mB1': 'PD1',\n",
    "            'mC1': 'PD1',\n",
    "            'mD1': 'RTIgG',\n",
    "            'mE1': 'RTPD1',\n",
    "            'mF1': 'RTPD1'\n",
    "        }\n",
    "\n",
    "adata.obs['treatment'] = adata.obs['sampleID'].apply(lambda x: tx_map[x])\n",
    "\n",
    "counts = adata.obs.groupby(by=['treatment', 'majority_voting']).count()['sampleID']\n",
    "\n",
    "total_per_cluster = counts['IgG', :] + counts['RTIgG', :]\n",
    "total = total_per_cluster.sum()\n",
    "\n",
    "f_exp_IgG = (total_per_cluster / total) * counts['IgG',:].sum()\n",
    "f_exp_RTIgG = (total_per_cluster / total) * counts['RTIgG',:].sum()\n",
    "\n",
    "result = scipy.stats.chisquare(np.array([counts['IgG', :], counts['RTIgG', :]]).T, \n",
    "                                np.array([f_exp_IgG, f_exp_RTIgG]).T,\n",
    "                                axis= 1)\n",
    "\n",
    "result = pd.DataFrame(result.pvalue, index=counts['IgG', :].index)\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "f_obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_per_cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
